src/Diagnostics.DataProviders/ConfigurationFactory.cs

```diff
@@ -37,9 +37,18 @@ public AppSettingsDataProviderConfigurationFactory(IHostingEnvironment env)
                 )
             );
 
-            builder.AddAzureKeyVault($"https://{builtConfig["DevKeyVaultName"]}.vault.azure.net/",
-                                     keyVaultClient,
-                                     new DefaultKeyVaultSecretManager());
+            if (env.IsProduction())
```

Comment:
You can set `ProdKeyVaultName` or `DevKeyVaultName` based on `IsProduction()` to avoid copying code (and the magic strings) here